### PR TITLE
Package srt.0.3.1

### PR DESCRIPTION
**Pull request by toots**
### `srt.0.3.1`
Binding for the Secure, Reliable, Transport protocol library
Secure Reliable Transport (SRT) is an open source transport technology
that optimizes streaming performance across unpredictable networks, such
as the Internet.
This package provides OCaml bindings to the C implementation library.

---
* Homepage: https://github.com/savonet/ocaml-srt
* Source repo: git+https://github.com/savonet/ocaml-srt.git
* Bug tracker: https://github.com/savonet/ocaml-srt/issues

---
:camel: Pull-request generated by opam-publish v2.3.0

### CODE REVIEW

**Comment by shonfeder** [comment]
Quite a few CI failures on different compiler versions. I'll mark this as needing reporter action.

**Comment by toots** [comment]
> Quite a few CI failures on different compiler versions. I'll mark this as needing reporter action.

Woo thanks, this needs review indeed.

**Comment by mseri** [comment]
Many revdeps are breaking. Should we add an upper bound on the broken version of liquidsoap?
```
# File "io/srt_io.ml", line 510, characters 19-33:
# 510 |       let events = Srt.Poll.uwait t.p ~max_fds ~timeout:conf_timeout#get in
#                          ^^^^^^^^^^^^^^
# Error: This function has type
#          Srt.Poll.t -> timeout:int -> Srt.Poll.event list
#        It is applied to too many arguments; maybe you forgot a `;'.
# make[1]: *** [Makefile:395: io/srt_io.cmx] Error 2
# make[1]: Leaving directory '/home/opam/.opam/4.14/.opam-switch/build/liquidsoap.2.1.3/src'
# make: *** [Makefile:10: all] Error 2
```
(this can be done in a separate PR before merging this one)

**Comment by mseri** [comment]
Not really, here it says that it conflicts with srt < 0.1.2: https://github.com/ocaml/opam-repository/blob/master/packages/liquidsoap/liquidsoap.2.0.0/opam#L128

This means that any version >= 0.1.2 is good. We need to add a new conflict to srt >= 0.3.1

**Comment by toots** [comment]
> Not really, here it says that it conflicts with srt < 0.1.2: https://github.com/ocaml/opam-repository/blob/master/packages/liquidsoap/liquidsoap.2.0.0/opam#L128
> 
> This means that any version >= 0.1.2 is good. We need to add a new conflict to srt >= 0.3.1

Woof my bad! That's what happens when you respond too early.. Just pushed the changes!

File: packages/liquidsoap/liquidsoap.2.0.0/opam [hunk 1]
```diff
@@ -125,7 +125,7 @@ conflicts: [
   "shine" {< "0.2.0"}
   "soundtouch" {< "0.1.9"}
   "speex" {< "0.4.0"}
-  "srt" {< "0.1.2"}
+  "srt" {< "0.1.2" & >= "0.3.0"}
```
Comment (mseri) [inline packages/liquidsoap/liquidsoap.2.0.0/opam:128]:
```suggestion
  "srt" {< "0.1.2" | >= "0.3.0"}
```

File: packages/srt/srt.0.3.1/opam [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
+dev-repo: "git+https://github.com/savonet/ocaml-srt.git"
```
Comment (mseri) [inline packages/srt/srt.0.3.1/opam:42]:
```suggestion
conflicts: [ "ocaml-option-bytecode-only" ]
dev-repo: "git+https://github.com/savonet/ocaml-srt.git"
```

for
```
#=== ERROR while compiling srt.0.3.1 ==========================================#
# context              2.4.0~alpha1~dev | linux/x86_32 | ocaml-base-compiler.5.2.0 | pinned(https://github.com/savonet/ocaml-srt/archive/refs/tags/v0.3.1.tar.gz)
# path                 ~/.opam/5.2/.opam-switch/build/srt.0.3.1
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p srt -j 31 @install
# exit-code            1
# env-file             ~/.opam/log/srt-7-d0799a.env
# output-file          ~/.opam/log/srt-7-d0799a.out
### output ###
# File "src/generator/dune", lines 1-28, characters 0-653:
#  1 | (rule
#  2 |  (targets gen_stubs.exe)
#  3 |  (deps
# ....
# 26 |    %{lib-private:srt.types:srt_types.cmxa}
# 27 |    %{lib-private:srt.stubs:srt_stubs.cmxa}
# 28 |    %{lib-private:srt.stubs:locked/.srt_stubs_locked.objs/native/srt_stubs_locked.cmx})))
# Error: No rule found for src/stubs/srt_stubs.cmxa
# File "src/generator/dune", lines 1-28, characters 0-653:
#  1 | (rule
#  2 |  (targets gen_stubs.exe)
#  3 |  (deps
# ....
# 26 |    %{lib-private:srt.types:srt_types.cmxa}
# 27 |    %{lib-private:srt.stubs:srt_stubs.cmxa}
# 28 |    %{lib-private:srt.stubs:locked/.srt_stubs_locked.objs/native/srt_stubs_locked.cmx})))
# Error: No rule found for src/types/srt_types.cmxa
# File "src/generator/dune", lines 1-28, characters 0-653:
#  1 | (rule
#  2 |  (targets gen_stubs.exe)
#  3 |  (deps
# ....
# 26 |    %{lib-private:srt.types:srt_types.cmxa}
# 27 |    %{lib-private:srt.stubs:srt_stubs.cmxa}
# 28 |    %{lib-private:srt.stubs:locked/.srt_stubs_locked.objs/native/srt_stubs_locked.cmx})))
# Error: No rule found for src/constants/srt_constants.cmxa
# File "src/generator/dune", lines 1-28, characters 0-653:
#  1 | (rule
#  2 |  (targets gen_stubs.exe)
#  3 |  (deps
# ....
# 26 |    %{lib-private:srt.types:srt_types.cmxa}
# 27 |    %{lib-private:srt.stubs:srt_stubs.cmxa}
# 28 |    %{lib-private:srt.stubs:locked/.srt_stubs_locked.objs/native/srt_stubs_locked.cmx})))
# Error: No rule found for
# src/stubs/locked/.srt_stubs_locked.objs/native/srt_stubs_locked.cmx
```

File: packages/liquidsoap/liquidsoap.2.0.0/opam (hunk 1 above)
Comment (toots) [reply packages/liquidsoap/liquidsoap.2.0.0/opam]: Thank you for that...

**Comment by mseri** [comment]
Thanks!

**Pull request merged**